lms/services/oauth1.py

```diff
@@ -6,7 +6,9 @@ class OAuth1Service:
 
     def __init__(self, _context, request):
         self.request = request
-        self.ai_getter_service = request.find_service(name="ai_getter")
+        self.application_instance = request.find_service(
+            name="application_instance"
+        ).get()
```

Comment:
Heh... guess what ma feedback is! Yeah, I think we probably want `self.shared_secret =` rather than keeping the whole application instance about.

Reply:
I think this actually needs to not even call `ApplicationInstanceService.get()` until the `get_client()` method, otherwise `OAuth1Service.__init__()` could raise `ConsumerKeyError` which'd cause `request.find_service(name="oauth1")` to raise `ConsumerKeyError` which I think would be unexpected and a likely source of bugs. Fixed that: https://github.com/hypothesis/lms/pull/2651/commits/d18f7485d44918ac9c5e90669fb35ebc781ede0e